### ISSUE TITLE
Weird `-Xexperimental` behavior

### ISSUE THREAD

**Issue by notxcain** [typelevel/scala#113]
Based on [discussion](https://gitter.im/typelevel/scala?at=57c726fd29ee4a670581023b) on gitter.
- The code below fails to compile with `-Xexperimental` enabled however compiles just fine with `-Ypartial-unification` and `-Yliteral-types`. Interesting that removing `case` or removing `bar` parameter makes it compile.

``` scala
trait ~>[F[_], G[_]]

sealed trait DSL[A]
sealed trait FooDSL[A] extends DSL[A]
sealed trait BarDSL[A] extends DSL[A]

case class Inter(foo: FooDSL ~> List, bar: BarDSL ~> List)
```
- As @milessabin found out there is misconfigured `experimentalSettings` property, that enables only `-Yliteral-types`.  

``` scala
protected def experimentalSettings = List[BooleanSetting](YliteralTypes)
```

**Comment by milessabin** [comment]
I can reproduce this with vanilla 2.11.8,

```
miles@tarski:notxcain% scala exp.scala                                                                                                     
warning: there were two feature warnings; re-run with -feature for details
one warning found
miles@tarski:notxcain% scala -Xexperimental exp.scala                                                                                      
/home/miles/tmp/scala/notxcain/exp.scala:7: error: type mismatch;
 found   : this.~>[this.FooDSL,List]
 required: this.~>[? >: this.FooDSL with this.BarDSL <: this.DSL, [+A]List[A]]

case class Inter(foo: FooDSL ~> List, bar: BarDSL ~> List)
           ^
one error found
miles@tarski:notxcain%
```

So it looks like the `-Xexperimental` flag is changing behaviour elsewhere.

**Comment by milessabin** [comment]
For now I've updated the README to warn against using `-Xexperimental` ... the problem is also present in 2.12.x and will need another upstream PR.

**Comment by milessabin** [comment]
Fixed in 2.12.0-RC2.
